js_modules/dagster-ui/packages/ui-core/src/runs/CapturedLogPanel.tsx

```diff
@@ -26,36 +29,110 @@ interface CapturedOrExternalLogPanelProps extends CapturedLogProps {
+      if (logCaptureInfo?.logManagerMetadata) {
+        const path =
+          ioType === 'stdout' ? logCaptureInfo.stdoutUriOrPath : logCaptureInfo.stderrUriOrPath;
+
+        try {
+          const metadata = JSON.parse(logCaptureInfo.logManagerMetadata);
+          switch (metadata.type) {
+            case 'AzureBlobComputeLogManager':
+              if (metadata.storage_account && metadata.container) {
+                return `az storage blob download --account-name ${metadata.storage_account} --container-name ${metadata.container} --name ${path}`;
+              }
+          }
+        } catch {
+          return undefined;
+        }
+      }
+      return undefined;
+    };
+
     const externalUrl =
       logCaptureInfo &&
       (props.visibleIOType === 'stdout'
         ? logCaptureInfo.externalStdoutUrl
         : logCaptureInfo.externalStderrUrl);
-    if (externalUrl) {
+    const uriOrPath =
+      logCaptureInfo &&
+      (props.visibleIOType === 'stdout'
+        ? logCaptureInfo.stdoutUriOrPath
+        : logCaptureInfo.stderrUriOrPath);
+    const shellCmd = getShellCmd(props.visibleIOType, logCaptureInfo);
+
+    const copy = useCopyToClipboard();
+    const onClickFn = async (key: string, value: string | undefined) => {
+      if (!value) {
+        return;
+      }
+      copy(value);
+      await showSharedToaster({
+        intent: 'success',
+        icon: 'done',
+        message: `${key} copied!`,
+      });
+    };
+    const onClickExternalUri = async () => onClickFn('Log artifact URI', uriOrPath);
+    const onClickShellCmd = async () => onClickFn('Shell command', shellCmd);
+
+    if (externalUrl || uriOrPath || shellCmd) {
       return (
-        <Box
-          flex={{direction: 'row', alignItems: 'center', justifyContent: 'center', gap: 1}}
-          background={Colors.tooltipBackground()}
-          style={{color: Colors.tooltipText(), flex: 1, minHeight: 0}}
-        >
-          View logs at
-          <a
-            href={externalUrl}
-            target="_blank"
-            rel="noreferrer"
-            style={{
-              color: Colors.tooltipText(),
-              textDecoration: 'underline',
-              marginLeft: 4,
-              marginRight: 4,
-            }}
-          >
-            {externalUrl}
-          </a>
-          <Icon name="open_in_new" color={Colors.tooltipText()} size={20} style={{marginTop: 2}} />
-        </Box>
+        <CapturedLogDataTable>
+          <tbody>
+            {externalUrl ? (
+              <tr>
+                <td>View logs</td>
+                <td>
+                  <Box flex={{direction: 'row', alignItems: 'center', gap: 8}}>
+                    <a href={externalUrl} target="_blank" rel="noreferrer">
+                      {externalUrl}
+                      <Icon name="open_in_new" style={{display: 'inline-block'}} />
+                    </a>
+                  </Box>
```

Comment:
I suggest putting the `Box` inside the `a`, since the icon is currently a bit misaligned with the text.

```
<a href="...">
  <Box ...>
    <span>{externalUrl}</span>
    <Icon ... />
  </Box>
</a>
```

Reply:
thanks for the review, should I keep the Box outside of it? ie:

```html
<td>
  <Box>
    <a>
       <Box>
          <span />
          <icon />
       </Box>
    </a>
   </Box>
<td>
```

Reply:
I don't think you should need to. Rendering in table cells can do some unexpected things, but if it looks fine without the wrapper then no need to include one.